semantics/language/common/expr/eval.k

```diff
@@ -41,19 +43,19 @@ module C-COMMON-EXPR-EVAL
      }*/
 
      // Lvalue conversion.
-     rule reval(nclv(Loc:SymLoc, T:Type)
+     rule reval(nclv(Loc::SymLoc, T::Type)
```

Comment:
as a matter of fact, putting something other than a SymLoc as the first argument of an nclv will *not* typecheck unless the user explicitly casts a term placed in that position. In practice as long as you don't do this you will get an error if you make a mistake. I forgo the checks in a number of places in this diff, but all of them represent cases where my analysis indicates that the type system is guaranteed to not be violated.

The performance improvements associated with the substantial reductions in computing type information present in this diff are sufficiently significant that I don't really have any intention of compromising on this point, but if you want to propose some other kind of mechanism by which we can cut down on time spent computing sort information in critical rules of the semantics, I am willing to listen.